File: tools/toggle_visibility.py
import json
import os

# Path to the .vscode/settings.json file
settings_path = os.path.join('.vscode', 'settings.json')

# Read the settings file
try:
    with open(settings_path, 'r') as file:
        settings = json.load(file)
except FileNotFoundError:
    print(f"Error: {settings_path} not found.")
    exit(1)
except json.JSONDecodeError:
    print(f"Error: {settings_path} contains invalid JSON.")
    exit(1)

# Toggle the values of files.exclude
if 'files.exclude' in settings:
    for key in settings['files.exclude']:
        settings['files.exclude'][key] = not settings['files.exclude'][key]

# Write the updated settings back to the file
try:
    with open(settings_path, 'w') as file:
        json.dump(settings, file, indent=2)
    print("File visibility toggled successfully!")
except Exception as e:
    print(f"Error writing to {settings_path}: {e}")

File: frontend/src/toggle-visibility.ts
import fs from 'fs';
import path from 'path';

// Path to the .vscode/settings.json file
const settingsPath = path.join(__dirname, '.vscode', 'settings.json');

// Read the settings file
fs.readFile(settingsPath, 'utf8', (err, data) => {
  if (err) {
    console.error('Error reading settings file:', err);
    return;
  }

  try {
    // Parse the JSON content
    const settings: { 'files.exclude'?: Record<string, boolean> } = JSON.parse(data);

    // Toggle the values of files.exclude
    if (settings['files.exclude']) {
      for (const key in settings['files.exclude']) {
        if (settings['files.exclude'].hasOwnProperty(key)) {
          settings['files.exclude'][key] = !settings['files.exclude'][key];
        }
      }
    }

    // Convert the updated settings back to JSON
    const updatedSettings = JSON.stringify(settings, null, 2);

    // Write the updated settings back to the file
    fs.writeFile(settingsPath, updatedSettings, 'utf8', (err) => {
      if (err) {
        console.error('Error writing settings file:', err);
        return;
      }
      console.log('File visibility toggled successfully!');
    });
  } catch (parseError) {
    console.error('Error parsing settings file:', parseError);
  }
});
